src/language-js/printer-estree.js

```diff
@@ -4075,16 +4075,21 @@ function printArgumentsList(path, options, print) {
       arg &&
       arg.type === "ArrowFunctionExpression" &&
       arg.params &&
-      arg.params.some(
-        param =>
-          param.type &&
-          param.type === "ObjectPattern" &&
-          param.properties &&
-          param.properties.some(
-            (property, i, properties) =>
-              i < properties.length - 1 &&
-              isNextLineEmpty(options.originalText, property, options)
-          )
+      arg.params.some(hasEmptyLineInObject)
+    );
+  }
+
+  function hasEmptyLineInObject(node) {
+    return (
+      node &&
+      node.type &&
```

Comment:
This line can be omitted because you have `node.type === "ObjectPattern"` on the next line.